diff --git a/include/env/ti/k3_dfu.env b/include/env/ti/k3_dfu.env
index 201529636c..d43c82998a 100644
--- a/include/env/ti/k3_dfu.env
+++ b/include/env/ti/k3_dfu.env
@@ -8,11 +8,11 @@ dfu_alt_info_mmc=
        sysfw.itb fat 1 1
 
 dfu_alt_info_emmc=
-       rawemmc raw 0 0x800000 mmcpart 1;
+       rawemmc raw 0 0x400000 mmcpart 1;
        rootfs part 0 1 mmcpart 0;
        tiboot3.bin.raw raw 0x0 0x400 mmcpart 1;
-       tispl.bin.raw raw 0x400 0x1000 mmcpart 1;
-       u-boot.img.raw raw 0x1400 0x2000 mmcpart 1;
+       tispl.bin.raw raw 0x400 0x8000 mmcpart 1;
+       u-boot.img.raw raw 0xC00 0x400 mmcpart 1;
        u-env.raw raw 0x3400 0x100 mmcpart 1;
        sysfw.itb.raw raw 0x3600 0x800 mmcpart 1